newsfragments/823.fixed.md

```diff
@@ -0,0 +1 @@
+Fix header_oracle deadlocking on history validation
```

Comment:
Release notes are primarily for end users. This release note requires a lot of insider knowledge, and even for us devs that have it, the note doesn't necessarily help remind us exactly when the deadlock happens.

Anyway, it's not really important to fix, because there aren't any active users, and all the developers who need to know are on calls that let us clarify. But it's useful for you to think about how you would describe the situation to a user who doesn't know anything about the architecture of the code or the internal specs for the network:

- What bug could they observe?
- In what circumstances would that bug appear?
- Simple explanation of what was going wrong and how it was resolved, but without referencing internal names or specs

As you might imagine, it's quite challenging to do that well. It takes a lot of practice. Now is a great time to start practicing, with very low stakes. Feel free to send me a draft in chat. There's no need to hold up merging the PR.

Reply:
Thank you for letting me know. I sent you a practice one in a dm. I am going to remove the release note then merge it cause anything that requires or does validation in Trin if it doesn't have the headerWithProof locally will fail. I will follow these practices going forward.